### PR TITLE
Bd okta 320481 okta user profiles

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Addition to Okta Developer Docs / Concepts: User Profiles 
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-320481](https://oktainc.atlassian.net/browse/OKTA-320481)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+title: User Profiles
+meta:
+  - name: description
+    content: A high-level, developer-focused overview of user profiles and the Okta Universal Directory.
+---
+
+# User Profiles
+
+A user profile in Okta is the data record where information about your user is stored. A typical user profile holds information, or attributes, like a user’s first name, last name, username, and email address. Users can be employees, customers, partners, or end-users of applications.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:10]:
```suggestion
A user profile in Okta is the data record where user information is stored. A typical user profile contains information, or attributes, such as a user’s first name, last name, username, and email address. Users can be employees, customers, partners, or end-users of applications.
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+---
+title: User Profiles
+meta:
+  - name: description
+    content: A high-level, developer-focused overview of user profiles and the Okta Universal Directory.
+---
+
+# User Profiles
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:8]:
```suggestion
# User profiles
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+A user profile in Okta is the data record where information about your user is stored. A typical user profile holds information, or attributes, like a user’s first name, last name, username, and email address. Users can be employees, customers, partners, or end-users of applications.
+
+The default Okta user profile has [31 user attributes](/docs/reference/api/users/#default-profile-properties), which can be customized based on client requirements. Additional custom attributes can be added to the user profile to support most client user needs. Custom profile attribute types further provide the ability to customize the user experience based on your org and application needs.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:12]:
```suggestion
The default Okta user profile has [31 user attributes](/docs/reference/api/users/#default-profile-properties) that you can customize based on client requirements. Additional custom attributes can be added to the user profile to support most client user needs. Custom profile attribute types enable you to customize the user experience even more, based on your org and application needs.
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+You can manage user profile design and customization for your org, as well as individual user updates, from the Admin Console or using specific APIs.
+
+See also [Users, Groups, and Profiles](https://help.okta.com/en/prev/Content/Topics/users-groups-profiles/usgp-main.htm).
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:16]:
```suggestion
See [Users, Groups, and Profiles](https://help.okta.com/en/prev/Content/Topics/users-groups-profiles/usgp-main.htm).
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+---
+
+# User Profiles
+
+A user profile in Okta is the data record where information about your user is stored. A typical user profile holds information, or attributes, like a user’s first name, last name, username, and email address. Users can be employees, customers, partners, or end-users of applications.
+
+The default Okta user profile has [31 user attributes](/docs/reference/api/users/#default-profile-properties), which can be customized based on client requirements. Additional custom attributes can be added to the user profile to support most client user needs. Custom profile attribute types further provide the ability to customize the user experience based on your org and application needs.
+
+You can manage user profile design and customization for your org, as well as individual user updates, from the Admin Console or using specific APIs.
+
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:15]: Remove hard return (line 15) so lines 14 and 16 is a paragraph as the cross-ref is relevant to line 14.

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+See also [Users, Groups, and Profiles](https://help.okta.com/en/prev/Content/Topics/users-groups-profiles/usgp-main.htm).
+
+## What is the Okta Universal Directory?
+
+The Okta Universal Directory is the service that stores all information on your users (user profiles) for your organization. Depending on the set up for your org, the Universal Directory can serve as the “single-source-of-truth” for your users.
+
+Additionally, the Universal Directory holds app user profiles, which define the attributes that applications require from individual users. For example, one of your applications might only need to know the user’s name as one string (for example, “John Doe”) while another application might require the user’s first and last names to be separate (for example,  “John” and “Doe”). Furthermore, some applications might store sensitive information, like a user’s address, while other applications do not. If you were to share the same user profile with each application, both applications would have access to data they might not need or be authorized to view. With Universal Directory, you can be sure that each application only gets the data that it needs.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:22]:
```suggestion
Additionally, the Universal Directory holds app user profiles, which define the attributes that applications require from individual users. For example, one of your applications may only need to know the user’s name as one string (for example, “John Doe”) while another application may require the user’s first and last names to be separate (for example,  “John” and “Doe”). Furthermore, some applications may store sensitive information, like a user’s address, while other applications do not. If you were to share the same user profile with each application, both applications would have access to data they may not need or be authorized to view. With Universal Directory, you can be sure that each application only gets the data that it needs.
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+The Universal Directory has a single Okta user profile for every user and an app user profile for each application. The user profile is the primary place for all user information to be stored and the app users profile is where application specific information is stored.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:24]:
```suggestion
The Universal Directory has a single Okta user profile for every user and an app user profile for each application. The user profile is the primary place for all user information to be stored, and the app users profile is where application specific information is stored.
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+You can manage the User profiles in the Universal Directory from the Admin Console or using the [User API](/docs/reference/api/users).
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:26]:
```suggestion
You can manage the user profiles in the Universal Directory from the Admin Console or using the [User API](/docs/reference/api/users).
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+You can manage the Apps user profiles in the Universal Directory from the Admin Console or using the [Apps API](/docs/reference/api/apps).
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:28]:
```suggestion
You can manage the Apps user profiles in the Universal Directory from the Admin Console or use the [Apps API](/docs/reference/api/apps).
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+### User mappings
+
+The Universal Directory, in addition to storing user profiles and app user profiles, maps data from one profile to another, which keeps data synchronized between all of your applications. A common example of this action is storing a user’s first and last name in the user profile and mapping that data to an app user profile. In this way, a single change to a field in a User Profile is reflected in all the applications that map to that field.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:32]:
```suggestion
In addition to storing user profiles and app user profiles, the Universal Directory maps data from one profile to another, which keeps data synchronized between all of your applications. A common example of this action is storing a user’s first and last name in the user profile and mapping that data to an app user profile. In this way, a single change to a field in a User Profile is reflected in all the applications that map to that field.
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+See also [About attribute mappings](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-about-attribute-mappings.htm)
+
+You can manage the Universal Directory mappings between profiles using the Admin Console or the [Mappings API](/docs/reference/api/mappings/).
+
+## User Profile types
+
+Okta has two basic user profile types that define a user in the Universal Directory: Okta user profile type and the app user profile type. The Okta user profile type is further composed of Group profile types and Custom profile types.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:40]:
```suggestion
Okta has two basic user profile types that define a user in the Universal Directory: Okta user profile type and app user profile type. The Okta user profile type is further composed of Group profile types and Custom profile types.
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+See also [About profile types](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-about-profiles.htm) and [About custom user types in Universal Directory](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-usertypes-about.htm)
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:42]:
```suggestion
See [About profile types](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-about-profiles.htm) and [About custom user types in Universal Directory](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-usertypes-about.htm)
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+
+### Okta user profile type
+
+The Okta user profile type defines the default user record used in the Universal Directory. The default user profile contains 31 attributes in accordance with [RFC System for Cross-Domain Identity Management: Core Schema](https://datatracker.ietf.org/doc/html/rfc7643#section-4.1) and can also be extended with custom attributes. To manage the default user profile, use the [Users API](/docs/reference/api/users) and review the [Profile object](/docs/reference/api/users/#profile-object) and the [User object](https://developer.okta.com/docs/reference/api/users/#user-object) for further information.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:46]:
```suggestion
The Okta user profile type defines the default user record used in the Universal Directory. The default user profile contains 31 attributes in accordance with the [RFC System for Cross-Domain Identity Management: Core Schema](https://datatracker.ietf.org/doc/html/rfc7643#section-4.1) and can also be extended with custom attributes. To manage the default user profile, use the [Users API](/docs/reference/api/users), and review the [Profile object](/docs/reference/api/users/#profile-object) and the [User object](https://developer.okta.com/docs/reference/api/users/#user-object) for further information.
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+
+#### Group profile type
+
+Okta groups simplify management of multiple users of the same type, see [About groups](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-about-groups.htm). The Group profile itself consists of attributes and can be defined and managed with the [Groups API](/docs/reference/api/groups/). See the [Group object](/docs/reference/api/groups/#group-object) and [Group attributes](/docs/reference/api/groups/#group-attributes) for further information.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:50]:
```suggestion
Okta groups simplify management of multiple users of the same type. See [About groups](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-about-groups.htm). The Group profile itself consists of attributes, and can be defined and managed with the [Groups API](/docs/reference/api/groups/). See the [Group object](/docs/reference/api/groups/#group-object) and [Group attributes](/docs/reference/api/groups/#group-attributes).
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+
+#### Custom profile type
+
+The custom user profile type is based on the Okta user profile type and is used to define different types of users, for example administrators, contractors, help desk, and so on. Similar to the default Okta profile, the custom user profile type contains 31 attributes and can be extended with custom attributes. See also [About custom user types in Universal Directory](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-usertypes-about.htm).
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:54]:
```suggestion
The custom user profile type is based on the Okta user profile type and is used to define different types of users, for example administrators, contractors, help desk, and so on. Similar to the default Okta profile, the custom user profile type contains 31 attributes and can be extended with custom attributes. See [About custom user types in Universal Directory](https://help.okta.com/en/prod/Content/Topics/users-groups-profiles/usgp-usertypes-about.htm).
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+
+The [User Types API](/docs/reference/api/user-types/) defines and manages the custom profile types.
+
+User profiles can only belong to one user profile type. The [User object property](/docs/reference/api/users/#user-properties) `type` defines the custom user profile (or default profile) the user is associated with.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:58]:
```suggestion
User profiles can only belong to one user profile type. The [User object property](/docs/reference/api/users/#user-properties) `type` defines the custom user profile (or default profile) that the user is associated with.
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+
+>**Note:** The default [Profile object property](/docs/reference/api/users/#default-profile-properties) `userType` is a user profile attribute and is not a reference to the default or custom profile type.
+
+### App user profile type
+
+The app user profile type defines the attributes available for a user of that application in the Universal Directory. The app user profile attributes are mapped to the user profile and determines the data that can be sent to or imported from an app. Similar to user profiles, the app user profile has base attributes and custom attributes. The available custom attributes, however, are determined by the application. You can manage the app user profile type with the [Apps API](/docs/reference/api/apps/). Review the [Application User Profile](/docs/reference/api/apps/#application-user-profile-object) object and the [Application User](/docs/reference/api/apps/#application-user-object) object for further details.
+
+## Universal Directory Schemas
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:66]: Should "Schemas" be initial lowercase?

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+
+Many different types of data can be stored in a user profile: strings, numbers, dates, lists, and so on. A schema is a description of what type of information is stored in a user profile. Each element in a schema is known as an “attribute” and each attribute has the following metadata, or properties:
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:68]: Style guide says: "Begin each item in a list with a capital letter unless there's a reason not to (for example, it’s a command that's always lowercase). 
strings > Strings

Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:68]:
```suggestion
Many different types of data can be stored in a user profile: strings, numbers, dates, lists, and so on. A schema is a description of what type of information is stored in a user profile. Each element in a schema is known as an “attribute” and each attribute has the following metadata or properties:
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+
+* **Data type:** What kind of data is being stored. Examples include, string, number, and Boolean.
+* **Display name:** A human readable label to be used in User Interfaces
+* **Variable name:** The machine readable identifier for the attribute
+* **Description:** A more in-depth description of what the attribute is for
+* **Enum:** If the attribute value comes from a fixed list of choices
+* **Attribute Length:** how long the value can be, as appropriate for the attribute’s data type
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:75]:
```suggestion
* **Attribute Length:** How long the value can be, as appropriate for the attribute’s data type
```

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+* **Attribute required:** If an attribute is required, Okta gives an error if the attribute isn’t included.
+
+Schemas define every user profile type: Okta default user profile, custom user profiles, group user profiles, and app user profiles. The [Schemas API](/docs/reference/api/schemas) manages operations for all user profiles. See [User Schema object](/docs/reference/api/schemas/#user-schema-object), [App User Schema object]( /docs/reference/api/schemas/#app-user-schema-object), and [Group Schema object](/docs/reference/api/schemas/#group-schema-object) for further details.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:78]:
```suggestion
Schemas define every user profile type: Okta default user profile, custom user profiles, group user profiles, and app user profiles. The [Schemas API](/docs/reference/api/schemas) manages operations for all user profiles. See [User Schema object](/docs/reference/api/schemas/#user-schema-object), [App User Schema object]( /docs/reference/api/schemas/#app-user-schema-object), and [Group Schema object](/docs/reference/api/schemas/#group-schema-object).
```

**Review by christinechoy-okta — COMMENTED** [review]
Left a few comments. It looks great, Brian!

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md (hunk 3 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:12]: I kept `, which` here; using `that` implies there may be some default user attributes that are not customizable. Everything else I changed as you suggested.

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md (hunk 5 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:15]: The xref is relevant to the entire section; it's not specific to line 16.

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md (hunk 4 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:16]: I guess I used the "see also" context here to guide folks to our existing User Profile conceptual content on help.okta.com. I'm not against removing "also" here, but maybe it's a larger discussion? Or let me know what you think. I'll leave it for now.

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md (hunk 18 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:68]: Is this for ordered and unordered lists, though? Maybe I can just recast sentence. How 'bout: "Many different types of data can be stored in a user profile such as string, numbers, dates, lists, and so on."

File: packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md (hunk 4 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/concepts/user-profiles/index.md:16]: Yes, good to leave it for now. I was referring to our style guide, which states a very simple cross-ref but perhaps doesn't cover all different sorts of scenarios.

**Review by christinechoy-okta — APPROVED** [review]

**Comment by brianduffield-okta** [comment]
@joelfranusic-okta reviewed Google Docs version: https://docs.google.com/document/d/1vUOpXVuMZTkl51GN2_FIUUI2fcxrhK0aLOn5JJGtVow/edit?usp=sharing

**Pull request merged**